### PR TITLE
Add extra tags for error handling in migrations

### PR DESCRIPTION
**Pull request by PineappleIOnic**
## What does this PR do?

Adds extra data to our sentry logs on migrations

## Checklist

- [X] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: src/Appwrite/Platform/Workers/Migrations.php [hunk 1]
```diff
@@ -291,20 +294,26 @@ protected function processMigration(Document $project, Document $migration, Log
...
+                    $simpleMessages[] = "[Destination] ['{$error->getResourceType()}:{$error->getResourceId()}'] '{$error->getMessage()}'";
                 }
 
-                $migrationDocument->setAttribute('errors', $errorMessages);
+                $migrationDocument->setAttribute('errors', $prettyMessages);
+                $log->addExtra('migrationErrors', json_encode($simpleMessages));
                 $this->updateMigrationDocument($migrationDocument, $projectDocument);
 
+                throw new Exception('Migration Failed');
                 return;
```
Comment (loks0n) [inline src/Appwrite/Platform/Workers/Migrations.php:317]: Can we remove the `return;`?

File: src/Appwrite/Platform/Workers/Migrations.php [hunk 2]
```diff
@@ -291,20 +294,26 @@ protected function processMigration(Document $project, Document $migration, Log
...
+                    $prettyMessages[] = "Error occurred while pushing '{$error->getResourceType()}:{$error->getResourceId()}' to destination with message: '{$error->getMessage()}'";
+                    $simpleMessages[] = "[Destination] ['{$error->getResourceType()}:{$error->getResourceId()}'] '{$error->getMessage()}'";
                 }
 
-                $migrationDocument->setAttribute('errors', $errorMessages);
+                $migrationDocument->setAttribute('errors', $prettyMessages);
+                $log->addExtra('migrationErrors', json_encode($simpleMessages));
                 $this->updateMigrationDocument($migrationDocument, $projectDocument);
 
+                throw new Exception('Migration Failed');
```
Comment (loks0n) [inline src/Appwrite/Platform/Workers/Migrations.php:316]:
```suggestion
                throw new Exception('Migration failed');
```

**Review by loks0n — APPROVED** [review]

**Comment by loks0n** [comment]
Should this be targeting `1.5.x`?

**Comment by PineappleIOnic** [comment]
Closing this to rebase it. Moving to #8055

**Pull request closed without merging**